src/lib/azure/azurecredentials.ts

```diff
@@ -85,13 +87,21 @@ export const getManagementCredentials = async (
     return undefined;
   }
 
-  // verifyConfigDefined has confirmed that these values are defined.
-  return msRestNodeAuth.loginWithServicePrincipalSecret(
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    servicePrincipalId!,
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    servicePrincipalPassword!,
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    tenantId!
-  );
+  try {
+    // verifyConfigDefined has confirmed that these values are defined.
+    return await msRestNodeAuth.loginWithServicePrincipalSecret(
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      servicePrincipalId!,
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      servicePrincipalPassword!,
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
Is there a better method to enforce non-null that does not require eslint-disable?

Reply:
i do not know. please let me know if you have some ideas

Reply:
@evanlouie thoughts?